Pzuraq reactivity blog posts (#529)
ember-learn/ember-blog: source/2020-02-14-the-ember-times-issue-135.md

---

## [Autotracking! A Blog Series on Reactivity Systems 🤖](https://www.pzuraq.com/thinking-with-autotracking-what-makes-a-good-reactive-system/)

> Interested in learning more about auto tracking, Ember Octane's new reactivity system? Well then, you're in luck 🍀. [Chris Garrett @pzuraq](https://github.com/pzuraq) is starting another super interesting new blog post series on that exact subject!

Reviewer: I removed the space for consistency.
```suggestion
Interested in learning more about autotracking, Ember Octane's new reactivity system? Well then, you're in luck 🍀. [Chris Garrett (@pzuraq)](https://github.com/pzuraq) is starting another super interesting new blog post series on that exact subject!
```

> The [first post](https://www.pzuraq.com/thinking-with-autotracking-what-makes-a-good-reactive-system/) came out at the end of last month and it examines what reactivity means, how it relates to declarative and imperative programming models. Even experienced Ember developers will likely get a lot of value out of reading about what these often-used, but rarely fully understood, terms mean. 

Reviewer: ```suggestion
The [first post](https://www.pzuraq.com/thinking-with-autotracking-what-makes-a-good-reactive-system/), which came out at the end of last month, examines what reactivity means and how it relates to declarative and imperative programming models. Even experienced Ember developers will likely get a lot of value out of reading about what these often-used, but rarely fully understood, terms mean. 
```

> Explaining things in an approachable and conversational style, [@pzuraq](https://github.com/pzuraq) also looks at how state is used in these programming models and how it is used in functional programming models. It’s a great intro for any one looking to learn a bit about reactivity systems in general.

Reviewer: ```suggestion
In an approachable and conversational style, [@pzuraq](https://github.com/pzuraq) also explains how state is used in these programming models as well as functional programming models. It’s a great intro for anyone looking to learn a bit about reactivity systems.
```

Bonus 🎉 [@pzuraq’s](https://github.com/pzuraq) [second post](https://www.pzuraq.com/thinking-with-autotracking-what-makes-a-good-reactive-system/) about what makes a good reactivity system came out this week!

> In this second post [@pzuraq](https://github.com/pzuraq) dives deeper and surveys several reactivity systems. Compaing HTML/CSS (did you know they are reactive!?), push-based reactivity systems that are often implemented with JavaScript (such as Ember Classic and other systems like the Observable model), and pull-based reactivity systems of React and the Virtual DOM, Vue and Elm. 

Reviewer: I found the second sentence a bit hard to parse. If I understood the sentence right, I think making a list like this may help:
```suggestion
[@pzuraq](https://github.com/pzuraq) dives deeper and surveys several reactivity systems:
- HTML/CSS (did you know they are reactive!?)
- Push-based reactivity systems, which are often implemented with JavaScript (e.g. Ember Classic and other systems like the Observable model)
- Pull-based reactivity systems (e.g. React and the Virtual DOM, Vue and Elm)
```

> There's some helpful diagrams and clear explanations to help you better understand the finer points. If you’re looking to learn more about how these models of reactivity are similar and different, then this post is a great read for you.

Reviewer: ```suggestion
There are helpful diagrams and clear explanations to help you better understand the finer points. If you’re looking to learn more about how these models of reactivity are similar and different, then this post is a great read for you.
```

> Be sure to be on the look out for more updates from the Ember Times team as more of these awesome blog posts are released!

Reviewer: ```suggestion
Be sure to be on the lookout for more updates from the Ember Times team as more of these awesome blog posts are released!
```
